### PR TITLE
TestStatusServer - add retry and report full errors.

### PR DESCRIPTION
**Pull request by BramGruneir**
First attempt at addressing #2059
I've changed the test slight to see if we can better diagnose the weird flaky behavior. 
- Added a util/retry to the get loop.
- When there is an error in the body of the response, it outputs it in the fail message.

The plan here is to detect what's causing the errors and then fix them.

### CODE REVIEW

File: acceptance/status_server_test.go [hunk 1]
```diff
@@ -31,30 +31,70 @@ import (
...
+	// There seems to be some issues while trying to connect to the status
+	// server, so retry (up to 5 times) with a 1 second delay each time.
+	// TODO(Bram): Clean this up once we get to the bottom of the issue.
+	for retry := 0; retry < 5; retry++ {
+		resp, err := client.Get(url)
+		if err != nil {
+			if retry < 4 {
+				t.Logf("could not GET %s - %s", url, err)
+				time.Sleep(100 * time.Millisecond)
+				client = getNewClient()
```
Comment (tamird) [inline acceptance/status_server_test.go]: why does this require a new client each time?

File: acceptance/status_server_test.go [hunk 2]
```diff
@@ -31,30 +31,70 @@ import (
...
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("could not GET %s - statuscode: %d", url, resp.StatusCode)
-	}
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		t.Fatalf("could not open body for %s - %s", url, err)
+	// There seems to be some issues while trying to connect to the status
+	// server, so retry (up to 5 times) with a 1 second delay each time.
+	// TODO(Bram): Clean this up once we get to the bottom of the issue.
+	for retry := 0; retry < 5; retry++ {
```
Comment (tamird) [inline acceptance/status_server_test.go]: looks like a fit for backoff

File: acceptance/status_server_test.go (hunk 1 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: There was an error with use of a closed connection.  I wanted to see if a new client would fix the issue.

File: acceptance/status_server_test.go (hunk 2 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: Is that a known golang pattern?  Or are you suggesting adding a larger backoff each time?

File: acceptance/status_server_test.go (hunk 1 above)
Comment (tamird) [inline acceptance/status_server_test.go]: this is a pretty blunt way to force a reconnect. go's http clients do not represent a single tcp connection; i wouldn't expect this to do anything

File: acceptance/status_server_test.go (hunk 2 above)
Comment (tamird) [inline acceptance/status_server_test.go]: util/retry

File: acceptance/status_server_test.go (hunk 1 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: I'll remove it.

File: acceptance/status_server_test.go (hunk 2 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: Ah, perfect!

**Comment by BramGruneir** [comment]
Updated and updated the pr description as well.

File: acceptance/status_server_test.go [hunk 3]
```diff
@@ -31,30 +31,45 @@ import (
...
+	"github.com/cockroachdb/cockroach/util/retry"
 )
 
 type details struct {
 	NodeID proto.NodeID `json:"nodeID"`
 }
 
+var retryOptions = retry.Options{
+	InitialBackoff: 100 * time.Millisecond,
+	MaxBackoff:     5 * time.Second,
```
Comment (tamird) [inline acceptance/status_server_test.go]: i think you should replace this line with `MaxRetries: 4,`

File: acceptance/status_server_test.go [hunk 4]
```diff
@@ -31,30 +31,45 @@ import (
...
+	// TODO(Bram): Clean this up once we get to the bottom of the issue.
+	for r := retry.Start(retryOptions); r.Next(); {
+		resp, err := client.Get(url)
+		if err != nil {
+			t.Logf("could not GET %s - %s", url, err)
+			continue
+		}
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Logf("could not open body for %s - %s", url, err)
```
Comment (tamird) [inline acceptance/status_server_test.go]: missing `continue` here?

**Comment by tamird** [comment]
LGTM

File: acceptance/status_server_test.go (hunk 4 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: added

File: acceptance/status_server_test.go (hunk 3 above)
Comment (BramGruneir) [inline acceptance/status_server_test.go]: done

**Pull request merged**